test/src/org/labkey/test/tests/targetedms/TargetedMSInstrumentNicknameTest.java

```diff
@@ -0,0 +1,258 @@
+
+    @Override
+    protected String getProjectName()
+    {
+        return getClass().getSimpleName() + " Project";
+    }
+
+    @BeforeClass
+    public static void initProject()
+    {
+        TargetedMSInstrumentNicknameTest init = getCurrentTest();
+        init.setupProjectWithSubfolders();
+        init.importInitialData();
+    }
+
+    @Override
+    protected void doCleanup(boolean afterTest) throws TestTimeoutException
+    {
+        super.doCleanup(afterTest);
+
+        // Clean out the nicknames set in /Shared
+        var command = new SelectRowsCommand("targetedms", "InstrumentNickname");
+        command.setFilters(List.of(new Filter("Nickname", AUTOMATED_TEST_NICKNAME_PREFIX, Filter.Operator.STARTS_WITH)));
+        try
+        {
+            Connection connection = createDefaultConnection();
+            var response = command.execute(connection, "/Shared");
+            if (!response.getRows().isEmpty())
+            {
+                var deleteCommand = new DeleteRowsCommand("targetedms", "InstrumentNickname");
+                deleteCommand.setRows(response.getRows());
+                deleteCommand.execute(connection, "/Shared");
+            }
+        }
+        catch (IOException | CommandException e)
+        {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private void setupProjectWithSubfolders()
+    {
+        setupFolder(FolderType.QC);
+
+        setupSubfolder(getProjectName(), QC_SUB_FOLDER, FolderType.QC);
+        setupSubfolder(getProjectName(), NON_QC_SUB_FOLDER, FolderType.Experiment);
+
+        clickFolder(QC_SUB_FOLDER);
+        setupSubfolder(getProjectName(), QC_SUB_FOLDER, QC_SUB_SUB_FOLDER, FolderType.QC);
+
+        _userHelper.createUser(USER);
+
+        // give user reader permissions to all but FOLDER_1
+        ApiPermissionsHelper permissionsHelper = new ApiPermissionsHelper(this);
+        permissionsHelper.addMemberToRole(USER, "Reader", PermissionsHelper.MemberType.user, getProjectName());
+        permissionsHelper.addMemberToRole(USER, "Editor", PermissionsHelper.MemberType.user, getProjectName() + "/" + NON_QC_SUB_FOLDER);
+    }
+
+    private void importInitialData()
+    {
+        goToProjectHome();
+        importData(ISOTOPOLOGUE_FILE_ANNOTATED, 1, false, false);
+
+        // Import the same file into a subfolders to test scoping
+        clickFolder(QC_SUB_FOLDER);
+        importData(ISOTOPOLOGUE_FILE_ANNOTATED, 1, false, false);
+
+        clickFolder(QC_SUB_SUB_FOLDER);
+        importData(ISOTOPOLOGUE_FILE_ANNOTATED, 1, false, false);
+
+        clickFolder(NON_QC_SUB_FOLDER);
+        importData(ISOTOPOLOGUE_FILE_ANNOTATED, 1, false, false);
+        // Only do DB maintenance on the last import in the sequence
+        importData(SAMPLE_FILE_CHROM_INFO, 2, false, true);
+    }
+
+    @Before
+    public void preTest()
+    {
+        goToProjectHome();
+    }
```

Comment:
This `preTest` is redundant since the individual tests `goToProjectHome`. I don't really have a preference but you don't need to do both.

Reply:
I removed `preTest()`